server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java

```diff
@@ -395,4 +266,178 @@ public String getViolationDescription(short violationCode) {
     return null;
   }
 
+  private void validateColValLen(ArrayList<Short> violations, ColumnUpdate columnUpdate) {
+    Text columnFamily = new Text(columnUpdate.getColumnFamily());
+    if (columnUpdate.getValue().length == 0 && !(columnFamily.equals(ScanFileColumnFamily.NAME)
+        || columnFamily.equals(LogColumnFamily.NAME))) {
+      addViolation(violations, 6);
+    }
+  }
+
+  private void validateTabletRow(ArrayList<Short> violations, byte[] row) {
+    // check the row, it should contain at least one ";" or end with "<". Row should also
+    // not be less than AccumuloTable.METADATA.tableId().
+    boolean containsSemiC = false;
+
+    for (byte b : row) {
+      if (b == ';') {
+        containsSemiC = true;
+      }
+
+      if (b == ';' || b == '<') {
+        break;
+      }
+
+      if (!validTableNameChars[0xff & b]) {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    if (!containsSemiC) {
+      // see if last row char is <
+      if (row.length == 0 || row[row.length - 1] != '<') {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    if (row.length > 0 && row[0] == '!') {
+      if (row.length < 3 || row[1] != '0' || (row[2] != '<' && row[2] != ';')) {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    // ensure row is not less than AccumuloTable.METADATA.tableId()
+    if (new Text(row).compareTo(new Text(AccumuloTable.METADATA.tableId().canonical())) < 0) {
+      addViolation(violations, 5);
+    }
```

Comment:
Noticed this could be done w/ Arrays.compare if we precompute the metadata table as a `byte[]` which would avoid allocating two text objs to do this comparison.  Could be a follow on PR

Reply:
Simple so added in this PR